### DISCUSSION TITLE
How to query fields inside custom json field

### DISCUSSION THREAD

**Discussion by subhashkoganti** [netbox-community/netbox#13286] [Q&A]
Does anybody know how to query the fields inside custom json field?

I know that we can use 'cf_{custom_field_name}' as query parameter to match the values we'd like to query.
However, I'm not able figure out the format if the custom field is of type 'json'.

Any help is appreciated.

I did search the existing discussions, but did not find anything related to custom json fields.

**Comment by candlerb** [comment]
AFAIK, it's not possible. The queries (for REST API) are documented [here](https://docs.netbox.dev/en/stable/reference/filtering/) and there's nothing for digging into of the body of a JSON custom field, other than "contains".

You could try GraphQL? Otherwise, you might need to restructure your custom fields, or do filtering client-side.

**Reply by subhashkoganti** [reply]
Thanks for your reply Brian.

I will check graphQL, as well, as you suggested.

I have a scenario where I need to store multiple data objects (discovered bgp peers) against a ASN object.
The only choice I could see was to use custom field with JSON type.

Even the contains filter expression - cf_{field_name}__ic - is not working (I'm using netbox v3.3.6)

**Reply by sirjeannot** [reply]
Same here on v3.5.9, the filter cf_{field_name}__ic - is not working, with the custom field defined as a JSON type.
